docs/_blog/2020-04-16-working-with-private-registries.md

```diff
@@ -0,0 +1,139 @@
+
+### A Node.js NPM Example
+
+I'm starting with a Codewind project created using the Node.js Express project template. When I open the application and navigate to its `/codewind` endpoint, I'm greeted with:
+
+![image of Hey there, this is codewind saying hi and thanks for using me](images/blog/privateregnode01.png){:width="500px"}
+*Hey there, this is codewind saying hi and thanks for using me*
+{:style="margin: 30px; text-align: center;"}
+
+The message looks pretty bland doesn't it? Luckily, I happen to have a special package hosted on my private GitHub NPM registry that I can use to give this message some flair. First, I'll configure the credentials for the registry in my `~/.npmrc` configuration file:
+
+```properties
+@makandre:registry=https://npm.pkg.github.com/makandre
+//npm.pkg.github.com/:_authToken=****************************************
+```
+{:style="margin: 30px;"}
+
+Now I can install the package to my project:
+
+```bash
+cwNodeProj$ npm install @makandre/samplepkg
+```
+{:style="margin: 30px;"}
+
+And now to use the package in the `cwNodeProj/server/routers/codewind.js` file:
+
+```js
+const express = require('express');
+const hello = require('@makandre/samplepkg'); // added require here
+
+module.exports = function (app) {
+  const router = express.Router();
+
+  router.get('/', function (req, res, next) {
+    // replaced "Hey there" with a call to hello()
+    const stringToReturn = hello() + ", this is codewind saying hi and thanks for using me"
+    res.status(200).send(stringToReturn);
+  });
+
+  app.use('/codewind', router);
+}
+```
+{:style="margin: 30px;"}
+
+All is set so I force a full project build in Codewind--right-click project in Codewind view and select *Build*... and the build fails. I check the build log and see the following:
+
+```bash
+npm ERR! code E404
+npm ERR! 404 Not Found: @makandre/samplepkg@^1.0.0
+```
+{:style="margin: 30px;"}
+
+So what happened? Well it's true I had configured everything correctly on my system to pull from the private registry. However, that's just for my local environment. When Codewind builds the project into a Docker image, the Docker build does not have access to the same configuration. This can be easily remedied by adding a [reference](referencing-files.html) to the `~/.npmrc` file into my project--using a reference keeps sensitive information out of my project where it may get accidentally committed to source control. After adding the reference, my project built successfully and it shows the updated message:
+
+*Hello there, general Kenobi, this is codewind saying hi and thanks for using me*
+{:style="margin: 30px; text-align: center;"}
```

Comment:
As much as I like Star Wars, I'm not sure if this would be a copyright issue or something? It might be safer to use a non-character name...or a character name that's also a "normal" name so that it's not an issue.

Reply:
That's no fun... but I'm fairly sure this is a case of "fair use" and besides, there are tons of  "Hello there" memes out there.  So I'm going to stand my (high) ground and insist (please?) to keep this reference.

Reply:
Then let's at least capitalize "General" as it's the name title. 😉 
I suppose we should also be capitalizing "Codewind," too?